### PR TITLE
hyperfine: add runtime dependency on bash.out

### PR DESCRIPTION
**Pull request by mgttlinger**
###### Description of changes
hyperfine calls `sh -c` at runtime and the `bash.out` derivation sets `sh` as alias for `bash`. Usually `sh` should be in the PATH from the OS already, however when using `nix bundle` you can run into situations where this is no longer given as there is no explicit dependency specified.

###### Things done

Added dependency on `bash.out`

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bjornfor** [comment]
Has https://github.com/matthewbauer/nix-bundle/issues/15 been fixed?

**Comment by mgttlinger** [comment]
> Has [matthewbauer/nix-bundle#15](https://github.com/matthewbauer/nix-bundle/issues/15) been fixed?

Not as far as I know but that issue is about the bundle itself requiring the host to have certain executables in place while here this is a case of hyperfine needing `sh` to run but not declaring the dependency, right?

**Comment by bjornfor** [comment]
> > Has [matthewbauer/nix-bundle#15](https://github.com/matthewbauer/nix-bundle/issues/15) been fixed?
> 
> Not as far as I know but that issue is about the bundle itself requiring the host to have certain executables in place while here this is a case of hyperfine needing `sh` to run but not declaring the dependency, right?

Right. I was just wondering since you said in the PR message:

> Usually sh should be in the PATH from the OS already, however when using nix bundle you can run into situations where this is no longer given as there is no explicit dependency specified.

**Comment by mgttlinger** [comment]
Yeah I mostly mentioned this as motivation for the change as you usually don't run into these kinds of issues. What I was doing is bundling `mkShell` equivalents where _only_ the dependencies are in PATH and hence not `sh` if nothing directly depends on `bash.out`.

**Review by bjornfor — APPROVED** [review]

File: pkgs/tools/misc/hyperfine/default.nix [hunk 1]
```diff
@@ -19,6 +20,8 @@ rustPlatform.buildRustPackage rec {
 
   nativeBuildInputs = [ installShellFiles ];
   buildInputs = lib.optional stdenv.isDarwin Security;
+  propagatedBuildInputs = [ bash.out # hyperfine uses 'sh -c' which gets symlinked as alias for bash by this derivation
+                          ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/hyperfine/default.nix:24]:
```suggestion
  propagatedBuildInputs = [
    # hyperfine uses 'sh -c' which gets symlinked as alias for bash by this derivation
    bash.out 
  ];
```

File: pkgs/tools/misc/hyperfine/default.nix [hunk 2]
```diff
@@ -19,6 +20,10 @@ rustPlatform.buildRustPackage rec {
 
   nativeBuildInputs = [ installShellFiles ];
   buildInputs = lib.optional stdenv.isDarwin Security;
+  propagatedBuildInputs = [
+    # hyperfine uses 'sh -c' which gets symlinked as alias for bash by this derivation
+    bash.out
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/hyperfine/default.nix:25]:
```suggestion
    bash.out
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/tools/misc/hyperfine/default.nix [hunk 3]
```diff
@@ -19,6 +20,10 @@ rustPlatform.buildRustPackage rec {
 
   nativeBuildInputs = [ installShellFiles ];
   buildInputs = lib.optional stdenv.isDarwin Security;
+  propagatedBuildInputs = [
+    # hyperfine uses 'sh -c' which gets symlinked as alias for bash by this derivation
```
Comment (Artturin) [inline pkgs/tools/misc/hyperfine/default.nix:24]: Using propagatedBuildInputs is wrong here, would it be possible to patch the path in src

**Comment by figsoda** [comment]
I'm not sure that we should do this, `sh` is only an optional dependency of hyperfine, and there is an easy way to opt out of using `sh`
```
  -S, --shell <SHELL>
          Set the shell to use for executing benchmarked commands. This can be
          the name or the path to the shell executable, or a full command line
          like "bash --norc". It can also be set to "default" to explicitly
          select the default shell on this platform. Finally, this can also be
          set to "none" to disable the shell. In this case, commands will be
          executed directly. They can still have arguments, but more complex
          things like "sleep 0.1; sleep 0.2" are not possible without a shell.
  -N
          An alias for '--shell=none'.
```

**Pull request closed without merging**